File: DuDoong-Infrastructure/src/main/java/band/gosrock/infrastructure/outer/api/dto/UnlinkKaKaoTarget.java
package band.gosrock.infrastructure.outer.api.dto;


import lombok.Getter;

@Getter
public class UnlinkKaKaoTarget {

    @feign.form.FormProperty("target_id_type")
    private String targetIdType = "user_id";

    @feign.form.FormProperty("target_id")
    private String aud;

    public UnlinkKaKaoTarget(String aud) {
        this.aud = aud;
    }

    public static UnlinkKaKaoTarget from(String aud) {
        return new UnlinkKaKaoTarget(aud);
    }
}

File: DuDoong-Infrastructure/src/main/kotlin/band/gosrock/infrastructure/outer/api/oauth/dto/UnlinkKaKaoTarget.kt
package band.gosrock.infrastructure.outer.api.oauth.dto

class UnlinkKaKaoTarget(aud: String) {

    @feign.form.FormProperty("target_id_type")
    val targetIdType: String = "user_id"

    @feign.form.FormProperty("target_id")
    val aud: String = aud

    companion object {
        @JvmStatic
        fun from(aud: String): UnlinkKaKaoTarget = UnlinkKaKaoTarget(aud)
    }
}
